src/query/management/src/role/role_mgr.rs

```diff
@@ -247,6 +249,55 @@ impl RoleApi for RoleMgr {
         Ok(Some(seq))
     }
 
+    /// Only drop role will call transfer.
+    ///
+    /// If a role is dropped, but the owner object is exists,
+    ///
+    /// The owner role need to update to account_admin.
+    #[async_backtrace::framed]
+    #[minitrace::trace]
+    async fn transfer_ownership_to_admin(
+        &self,
+        objects: &[(u64, OwnershipObject)],
+    ) -> databend_common_exception::Result<()> {
+        // Ensure accurate matching of a key
+        let mut if_then = vec![];
+        let mut condition = vec![];
+        for (seq, object) in objects {
+            let owner_key = self.ownership_object_key(object);
+            let owner_value = serialize_struct(
+                &OwnershipInfo {
+                    object: object.clone(),
+                    role: "account_admin".to_string(),
+                },
+                ErrorCode::IllegalUserInfoFormat,
+                || "",
+            )?;
+            condition.push(txn_cond_seq(&owner_key, Eq, *seq));
+            if_then.push(txn_op_put(&owner_key, owner_value))
+        }
+        let mut trials = txn_backoff(None, func_name!());
+        loop {
+            trials.next().unwrap()?.await;
+            let txn_req = TxnRequest {
```

Comment:
You did not re-fetch the `(seq, object)` thus this CAS loop has no effect at all.

You must re-fetch the latest `(seq, object)` in each loop.